torch/fx/experimental/symbolic_shapes.py

```diff
@@ -5482,3 +5483,133 @@ def run_node(self, n: torch.fx.Node):
         result = super().run_node(n)
         rebind_unbacked(detect_fake_mode().shape_env, n, result)
         return result
+
+
+def _parse_data_dependent_expr(e):
+    def _floor(x):
+        # sympy parsing transforms something like `a // b` into something like
+        # `sympy.floor((1/b) * a)`, which we need to transform back to `a // b`
+        if isinstance(x, sympy.Mul):
+            a, b = x.args
+            if isinstance(a, sympy.Rational):
+                return FloorDiv(a.p * b, a.q)
+        return sympy.floor(x)
```

Comment:
And if you retain the orig expression you don't need this fixer

Reply:
Sorry, missed this message before landing. Follow up here: #131932